src/jit/lower.cpp

```diff
@@ -462,6 +462,17 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
             hiOp2->gtNext      = hiResult;
             hiResult->gtPrev   = hiOp2;
 
+            if (oper == GT_ADD || oper == GT_SUB)
+            {
+                if (loResult->gtOverflow())
+                {
+                    hiResult->gtFlags |= GTF_OVERFLOW;
+                    if (loResult->gtFlags & GTF_UNSIGNED)
+                    {
+                        hiResult->gtFlags |= GTF_UNSIGNED;
+                    }
+                }
```

Comment:
It would be good to understand why this would cause another test to fail - it seems that we would want to always propagate it, if present.